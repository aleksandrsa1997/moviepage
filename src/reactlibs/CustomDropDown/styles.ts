import {createStyles, makeStyles, Theme, withStyles} from '@material-ui/core/styles';
import InputBase from '@material-ui/core/InputBase';
import colorMatrix from 'src/constants/colorMatrix';
import fonts from 'src/themes/fonts';

export const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        margin: {
            margin: theme.spacing(1),
        },
    }),
);

export const BootstrapInput = withStyles((theme: Theme) =>
    createStyles({
        root: {
            'label + &': {
                marginTop: theme.spacing(3),
            },
        },
        input: {
            borderRadius: 4,
            position: 'relative',
            backgroundColor: colorMatrix.colorBackground,
            border: `1px solid ${colorMatrix.shadow}`,
            fontSize: 16,
            width: 200,
            padding: '10px 26px 10px 12px',
            transition: theme.transitions.create(['border-color', 'box-shadow']),
            fontFamily: fonts.familyFont.join(','),
            '&:focus': {
                borderRadius: 4,
                borderColor: colorMatrix.colorBorder,
                boxShadow: colorMatrix.shadowBox,
            },
        },
    }),
)(InputBase);