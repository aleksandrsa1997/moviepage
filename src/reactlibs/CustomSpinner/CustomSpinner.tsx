import React from 'react';
import { CircularProgress } from '@material-ui/core';

export type TProps = {
    size?: number;
    value?: number;
    color?: 'inherit' | 'primary' | 'secondary';
    variant?: 'determinate' | 'indeterminate' | 'static';
    thickness?: number;
    disableShrink?: boolean;
};

const CustomSpinner: React.FC<TProps> = ({
    size = 40,
    color = 'primary',
    variant = 'determinate',
    thickness = 3.6
}) => (
    <CircularProgress
        size={size}
        color={color}
        variant={variant}
        thickness={thickness}
    />
);

export default CustomSpinner;