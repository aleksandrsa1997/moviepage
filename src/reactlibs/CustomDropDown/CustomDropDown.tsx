import FormControl from '@material-ui/core/FormControl';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import React from 'react';
import { TOption, TTempOption } from 'src/store/commonTypes';
import { useStyles } from './styles';

type TProps = {
    value?: string;
    mapKey?: string;
    options: Array<TOption | TTempOption>;
    onChange: (event: React.ChangeEvent<{ name?: string | undefined; value: unknown; }>) => void;
}

const CustomDropDown: React.FC<TProps> = ({
    value,
    mapKey,
    options,
    onChange,
}) => {
    const classes = useStyles();

    return (
        <FormControl className={classes.margin}>
            <Select
                id="demo-customized-select"
                labelId="demo-customized-select-label"
                onChange={onChange}
                value={value}
            >
                {
                    // @ts-ignore
                    options.map((item: any) => (
                            // @ts-ignore
                      <MenuItem key={item[mapKey]} value={item[mapKey]}>
                          {
                              // @ts-ignore
                              item[mapKey]
                          }
                        </MenuItem>
                    ),
                    )
                }
          </Select>
      </FormControl>
    );
};

export default CustomDropDown;
