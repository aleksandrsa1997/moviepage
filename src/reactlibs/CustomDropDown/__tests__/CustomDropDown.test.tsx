import React from 'react';
import CustomDropDown from 'src/reactlibs/CustomDropDown';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { TProps } from '../../CustomSpinner/CustomSpinner';

describe('CustomDropDown snapshot', () => {
    let props: TProps;
    let component: React.FC;

    beforeEach(()=> {
        props = {
            value: 'name',
            mapKey: 'name',
            options: [
                {
                    id: 0,
                    name: 'name',
                    adult: false,
                    gender: 1,
                    popularity: 1,
                    profile_path: 'profile_path',
                    known_for_department: 'known_for_department',
                }
            ],
            onChange: jest.fn()
        }
        component = mount(<CustomDropDown {...props} />);
    })

    it('should correct renderer', () => {

        expect(component).toMatchSnapshot();
    });

    it('should call onChange function ', () => {
        act(() => {
            component.find('CustomDropDown').at(0).getElement().props.onChange();
        })

        component.update();
        expect(props.onChange).toHaveBeenCalled();
    });
});