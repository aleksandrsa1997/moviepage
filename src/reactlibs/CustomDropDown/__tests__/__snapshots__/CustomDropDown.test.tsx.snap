// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomDropDown snapshot should correct renderer 1`] = `ReactWrapper {}`;
